packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts

```diff
@@ -775,6 +793,168 @@ export class SafeDsPythonGenerator {
             },
             { separator: '' },
         )!;
+        // Non-memoizable calls can be directly generated
+        if (!this.isCallMemoizable(expression)) {
+            return generatedPythonCall;
+        }
+        frame.addImport({ importPath: RUNNER_SERVER_PIPELINE_MANAGER_PACKAGE });
+        const hiddenParameters = this.getMemoizedCallHiddenParameters(expression, frame);
+        const callable = this.nodeMapper.callToCallable(expression);
+        const memoizedArgs = getParameters(callable).map(
+            (parameter) => this.nodeMapper.callToParameterValue(expression, parameter)!,
+        );
+        return expandTracedToNode(
+            expression,
+        )`${RUNNER_SERVER_PIPELINE_MANAGER_PACKAGE}.runner_memoized_function_call("${this.generateFullyQualifiedFunctionName(
+            expression,
+            frame,
+        )}", lambda *_ : ${generatedPythonCall}, [${joinTracedToNode(expression.argumentList, 'arguments')(
+            memoizedArgs,
+            (arg) => this.generateExpression(arg, frame),
+            { separator: ', ' },
+        )}], [${joinToNode(hiddenParameters, (param) => param, { separator: ', ' })}])`;
+    }
+
+    private isCallMemoizable(expression: SdsCall): boolean {
```

Comment:
`isMemoizableCall` would be more consistent with the other methods like `generateMemoizedCall`.